models.go

```diff
@@ -38,20 +40,67 @@ type pkbnPingResponse struct {
 
 type pkbnCreateResponse struct {
 	pkbnResponseStatus
-	// TODO contact support endpoint isn't returning the ID despite it being in their docs.
-	// ID string `json:"id"`
 }
 
-func (record pkbnRecord) toLibdnsRecord(zone string) libdns.Record {
-	ttl, _ := time.ParseDuration(record.TTL + "s")
-	priority, _ := strconv.Atoi(record.Prio)
-	return libdns.Record{
-		ID:       record.ID,
-		Name:     libdns.RelativeName(record.Name, LibdnsZoneToPorkbunDomain(zone)),
-		Priority: uint(priority),
-		TTL:      ttl,
-		Type:     record.Type,
-		Value:    record.Content,
+func (record pkbnRecord) toLibdnsRecord(zone string) (libdns.Record, error) {
+	name := libdns.RelativeName(record.Name, zone)
+	ttl, err := time.ParseDuration(record.TTL + "s")
+	if err != nil {
+		return libdns.RR{}, err
+	}
+
+	switch record.Type {
+	case "A", "AAAA":
+		ip, err := netip.ParseAddr(record.Content)
+		if err != nil {
+			return libdns.Address{}, err
+		}
+		return libdns.Address{
+			Name: name,
+			TTL:  ttl,
+			IP:   ip,
+		}, nil
+	case "CAA":
+		contentParts := strings.SplitN(record.Content, " ", 3)
+		flags, err := strconv.Atoi(contentParts[0])
+		if err != nil {
+			return libdns.CAA{}, err
+		}
+		tag := contentParts[1]
+		value := contentParts[2]
+		return libdns.CAA{
+			Name:  name,
+			TTL:   ttl,
+			Flags: uint8(flags),
+			Tag:   tag,
+			Value: value,
+		}, nil
+	case "CNAME":
+		return libdns.CNAME{
+			Name:   name,
+			TTL:    ttl,
+			Target: record.Content,
+		}, nil
+	case "SRV":
+		priority, err := strconv.Atoi(record.Prio)
+		return libdns.SRV{
+			Service:   "",
+			Transport: "",
+			Name:      name,
+			TTL:       ttl,
+			Priority:  uint16(priority),
+			Weight:    0,
+			Port:      0,
+			Target:    "",
+		}, err
```

Comment:
We should really parse this properly, I see for example, Cloudflare module does this

Splitting host gets Service and the Transport
Splitting the Value gets Weight, Port & Target

https://github.com/libdns/cloudflare/blob/master/models.go#L168-L182

Reply:
I think we might also need to add a method to handle parsing these libdns record back to Porkbun payloads properly. The current code wouldn't provision the SRV record correctly, which is mostly down to my naive initial implementation.

A good example of this kind of parsing here:

https://github.com/libdns/cloudflare/blob/master/models.go#L205-L246

Reply:
Woops, I had actually meant to delete this, but I wasn't aware caddy made use of them. I'll fix it

Reply:
Converting a porkbun record to libdns record is handled in [9d132fc](https://github.com/libdns/porkbun/pull/14/commits/9d132fc186150ea6783f863d7445008ee3a266cd), and vice versa in [e4f52ea](https://github.com/libdns/porkbun/pull/14/commits/e4f52ea76c6fe9a5224e528ba0aa57096af60f1e)